python/cudf/cudf/core/reshape.py

```diff
@@ -1160,3 +1160,271 @@ def _length_check_params(obj, columns, name):
+    """
+    if names is None:
+        names = []
+        for i, arr in enumerate(arrs):
+            if isinstance(arr, cudf.Series) and arr.name is not None:
+                names.append(arr.name)
+            else:
+                names.append(f"{prefix}_{i}")
+    else:
+        if len(names) != len(arrs):
+            raise ValueError("arrays and names must have the same length")
+        if not isinstance(names, list):
+            names = list(names)
+
+    return names
+
+
+def crosstab(
+    index,
+    columns,
+    values=None,
+    rownames=None,
+    colnames=None,
+    aggfunc=None,
+    margins=None,
+    margins_name=None,
+    dropna=None,
+    normalize=None,
+):
+    """
+    Compute a simple cross tabulation of two (or more) factors. By default
+    computes a frequency table of the factors unless an array of values and an
+    aggregation function are passed.
+
+    Parameters
+    ----------
+    index : array-like, Series, or list of arrays/Series
+        Values to group by in the rows.
+    columns : array-like, Series, or list of arrays/Series
+        Values to group by in the columns.
+    values : array-like, optional
+        Array of values to aggregate according to the factors.
+        Requires `aggfunc` be specified.
+    rownames : sequence, default None
+        If passed, must match number of row arrays passed.
+    colnames : sequence, default None
+        If passed, must match number of column arrays passed.
+    aggfunc : function, optional
+        If specified, requires `values` be specified as well.
+    margins : Not supported
+    margins_name : Not supported
+    dropna : Not supported
+    normalize : Not supported
+
+    Returns
+    -------
+    DataFrame
+        Cross tabulation of the data.
+
+    Examples
+    --------
+    >>> a = cudf.Series(["foo", "foo", "foo", "foo", "bar", "bar",
+    ...               "bar", "bar", "foo", "foo", "foo"], dtype=object)
+    >>> b = cudf.Series(["one", "one", "one", "two", "one", "one",
+    ...               "one", "two", "two", "two", "one"], dtype=object)
+    >>> c = cudf.Series(["dull", "dull", "shiny", "dull", "dull", "shiny",
+    ...               "shiny", "dull", "shiny", "shiny", "shiny"],
+    ...              dtype=object)
+    >>> cudf.crosstab(a, [b, c], rownames=['a'], colnames=['b', 'c'])
+    b   one        two
+    c   dull shiny dull shiny
+    a
+    bar    1     2    1     0
+    foo    2     2    1     2
+    """
+    if normalize is not None:
+        raise NotImplementedError("normalize is not supported yet")
+
+    if values is None and aggfunc is not None:
+        raise ValueError("aggfunc cannot be used without values.")
+
+    if values is not None and aggfunc is None:
+        raise ValueError("values cannot be used without an aggfunc.")
+
+    if not cudf.api.types.is_nested_list_like(index):
+        index = [index]
+    if not cudf.api.types.is_nested_list_like(columns):
+        columns = [columns]
```

Comment:
I don't think `is_nested_list_like` is the right utility to use here. Pandas uses it to check if you're  passing something like a list of Pandas Series objects:

```python
In [7]: pd.core.dtypes.inference.is_nested_list_like([pd.Series([1, 2, 3])])
Out[7]: True
```

But that will also return `True` if you have a list of list of lists for  example:

```python
In [8]: pd.core.dtypes.inference.is_nested_list_like([[[[1]]]])
Out[8]: True
```

I think to start with, we can keep this really simple, 

```python
if not isinstance(index, (list, tuple)):
    # assume the user passed a *single* Series-like object
    index = [index] 
```

We do this in other places as well.

Reply:
Agree. Have made relevant changes.